subxt/src/utils/unchecked_extrinsic.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! The "default" Substrate/Polkadot UncheckedExtrinsic.
+//! This is used in codegen for runtime API calls.
+//!
+//! The inner bytes represent the encoded extrinsic expected by the
+//! runtime APIs. Deriving `EncodeAsType` would lead to the inner
+//! bytes to be re-encoded (length prefixed).
+
+use std::marker::PhantomData;
+
+use codec::{Decode, Encode};
+use scale_decode::{visitor::DecodeAsTypeResult, IntoVisitor, Visitor};
+
+use super::{Encoded, Static};
+
+/// The unchecked extrinsic from substrate.
+#[derive(Clone, Debug, Eq, PartialEq, Encode, Decode, scale_encode::EncodeAsType)]
```

Comment:
Ah alas we need to decalre it ourselves because the derive `EncodeAsType` impl will try to be smarter and make sure it's encoding this whole thing as a tuple, where we just want to defer to the `Static` impl  and ignore this surrounding type entirely (this is why the test is failing)

Reply:
Yep exactly, we could at least remove the manual `Encode` part. I do wonder, could we make the `EncodeAsType` somehow ignore any fields that have `#[codec(skip)]` on them (similar to the codec)? 🤔

Reply:
And good idea on using `Encode` and `Decode` derives; that helps!